packages/backend/src/bitbucket.ts

```diff
@@ -0,0 +1,643 @@
+    logger.info(`Found ${workspaces.length} workspaces in ${workspaceMs} ms`);
+
+    for (const workspace of workspaces) {
+        const { durationMs: projectMs, data: projects } = await measure(() => client.getProjects(client, workspace));
+        logger.info(`  Found ${projects.length} projects in ${workspace} in ${projectMs} ms`);
+
+        for (const project of projects) {
+            const { durationMs, data: repos } = await measure(() => client.getRepos(client, workspace, project));
+            logger.info(`    Found ${repos.length} repos in ${project} in ${durationMs} ms`);
+
+            for (const repo of repos) {
+                logger.debug('REPO ' + JSON.stringify(repo));
+
+                var stars: number = 0;
+                var forks: number = 0;
+
+                // counting these can cause rate limiting
+                if (config.countMisc) {
+                    // not sure the suggested mapping of watchers to stars makes much sense
+                    const { durationMs: starMs, data: foundStars } = await measure(() => client.countWatchers(client, repo));
+                    stars = foundStars
+                    logger.info(`    Found ${stars} watchers for ${repo.name} in ${starMs} ms`);
+                    const { durationMs: forkMs, data: foundForks } = await measure(() => client.countForks(client, repo));
+                    forks = foundForks
+                    logger.info(`    Found ${forks} forks for ${repo.name} in ${forkMs} ms`);
+                }
+
+                repo.forks = forks;
+                repo.stars = stars;
+                repo.token = client.token ? client.token : UNKNOWN;
+                repo.user = client.user ? client.user : UNKNOWN;
+
+                foundRepos.push(repo);
+            }
+        }
+    };
+
+    let repos: GitRepository[] = foundRepos
+        .map((project) => {
+            const repoId = `${client.gitUrl}/${project.workspace}/${project.name}`;
+            const repoPath = path.resolve(path.join(ctx.reposPath, `${repoId}.git`))
+            const webUrl = new URL(repoId); // we don't want user & token here
+
+            const cloneUrl = new URL(repoId);
+            if (project.user) {
+                cloneUrl.username = project.user;
+            }
+            if (project.token) {
+                cloneUrl.password = project.token;
+            }
+
+            return {
+                vcs: 'git',
+                codeHost: 'bitbucket',
+                name: `${project.workspace}/${project.name}`,
+                id: repoId,
+                cloneUrl: cloneUrl.toString(),
+                path: repoPath,
+                isStale: false,
+                isFork: project.isFork,
+                isArchived: project.isArchived,
+                gitConfigMetadata: {
+                    //
+                    // using 'bitbucket-server' lets 'zoekt' generate the correct commit
```

Comment:
To confirm, the URL pattern for bitbucket cloud is the same as bitbucket server?
Looking at [the zoekt code](https://github.com/sourcegraph/zoekt/blob/dc1b23bb0da9a0fa91056ad6d5fcf9b8e641f67d/gitindex/index.go#L146), it looks like the URL pattern is:
```javascript
// Commit URL:
// https://<bitbucketserver-host>/projects/<project>/repos/<repo>/commits/5be7ca73b898bf17a08e607918accfdeafe1e0bc

// File URL:
// https://<bitbucketserver-host>/projects/<project>/repos/<repo>/browse/<file>?at=5be7ca73b898bf17a08e607918accfdeafe1e0bc
```